mesop/examples/llm_rewriter.py

```diff
@@ -0,0 +1,404 @@
+  input: str
+  output: list[ChatMessage]
+  in_progress: bool
+  rewrite: str
+  rewrite_message_index: int
+  preview_rewrite: str
+  preview_original: str
+  modal_open: bool
+
+
+@me.page(path="/llm-rewriter", title="LLM Rewriter")
+def page():
+  state = me.state(State)
+
+  # Modal
+  with me.box(style=_make_modal_background_style(state.modal_open)):
+    with me.box(style=_STYLE_MODAL_CONTAINER):
+      with me.box(style=_STYLE_MODAL_CONTENT):
+        me.textarea(
+          key="preview_rewrite",
+          label="Rewrite",
+          style=_STYLE_INPUT_WIDTH,
+          value=state.rewrite,
+          on_input=on_input_update(state),
+        )
+        with me.box():
+          me.button(
+            "Submit Rewrite",
+            color="primary",
+            type="flat",
+            on_click=on_click_submit_rewrite,
+          )
+          me.button(
+            "Cancel",
+            on_click=on_click_cancel_rewrite,
+          )
+        with me.box(style=_STYLE_PREVIEW_CONTAINER):
+          with me.box(style=_STYLE_PREVIEW_ORIGINAL):
+            me.text("Original Message", type="headline-6")
+            me.markdown(state.preview_original)
+
+          with me.box(style=_STYLE_PREVIEW_REWRITE):
+            me.text("Preview Rewrite", type="headline-6")
+            me.markdown(state.preview_rewrite)
+
+  # Chat UI
+  with me.box(style=_STYLE_APP_CONTAINER):
+    with me.box(style=_make_style_chat_ui_container(bool(_TITLE))):
+      me.text(_TITLE, type="headline-5", style=_STYLE_TITLE)
+      with me.box(style=_STYLE_CHAT_BOX):
+        for index, msg in enumerate(state.output):
+          with me.box(
+            style=_make_style_chat_bubble_wrapper(msg.role),
+            key=f"msg-{index}",
+            on_click=on_click_rewrite_msg,
+          ):
+            if msg.role == _ROLE_ASSISTANT:
+              me.text(
+                _display_username(_BOT_USER_DEFAULT, msg.edited),
+                style=_STYLE_CHAT_BUBBLE_NAME,
+              )
+            with me.box(style=_make_chat_bubble_style(msg.role, msg.edited)):
+              if msg.role == _ROLE_USER:
+                me.text(msg.content, style=_STYLE_CHAT_BUBBLE_PLAINTEXT)
+              else:
+                me.markdown(msg.content)
+                with me.tooltip(message="Rewrite response"):
+                  me.icon(icon="edit_note")
+
+      with me.box(style=_STYLE_CHAT_INPUT_BOX):
+        me.input(
+          label=_LABEL_INPUT,
+          # Workaround: update key to clear input.
+          key=f"input-{len(state.output)}",
+          on_input=on_input_update(State),
+          style=_STYLE_CHAT_INPUT,
+        )
+        with me.box():
+          me.button(
+            _LABEL_BUTTON_IN_PROGRESS if state.in_progress else _LABEL_BUTTON,
+            color="primary",
+            type="flat",
+            disabled=state.in_progress,
+            on_click=on_click_submit_chat_msg,
+          )
+
+
+# Event Handlers
+
+
+def on_input_update(State):
```

Comment:
even though this works, imo, the code is harder to understand vs the naive approach of separate handlers, if someone wants to fork this example

Reply:
Great point. Updated